docs/extensibility/theme-support.md

```diff
@@ -8,7 +8,7 @@ To opt-in for one of these features, call `add_theme_support` in the `functions.
 
 ```php
 function mytheme_setup_theme_supported_features() {
-	add_theme_support( 'editor-color-palette',
+	add_theme_support( 'editor-color-palette', array(
```

Comment:
Do I guess the usage right, that you want to create the array somewhat programatically and just pass a variable to `add_theme_support`? 

All in all I think it is a good method of doing things.

Reply:
@Luehrsen I won't speak for @webmandesign but that's what we'd like to do. Also makes it a little easier to re-use the color values in other contexts, e.g. auto-generating inline CSS for the classes, etc.

(something like…

```
$colors = [ /* imagine I'm an array */ ];

add_theme_support( 'editor-color-palette', $colors );

foreach ( $colors as $color ) {
}

Reply:
Although I like the approach, I would probably have done that with `get_theme_support()`.  I think that would add consistency throughout the themes and coding standards, as I would just have to search for 'editor-color-palette' to see all code that handles the theme colors.

```
foreach( get_theme_support( 'editor-color-palette' ) as $color ) {
     $css .= ".{$color['name'} { color: {$color['color']} }\n"
}
```

Reply:
@Luehrsen That works as well! My code was just an example — but having a consistent input and output format makes sense I think, either way.

Reply:
@Luehrsen Yes, that's correct, I want to create the array programatically and then just pass it into `add_theme_support`. I think that's the best approach (right now I can't think of any other `add_theme_support` feature being passed with multiple arguments nowadays, actually).